core/crypto/primitives/aes.go

```diff
@@ -39,108 +38,101 @@ func GenAESKey() ([]byte, error) {
 	return GetRandomBytes(AESKeyLength)
 }
 
-// PKCS7Padding pads as prescribed by the PKCS7 standard
-func PKCS7Padding(src []byte) []byte {
-	padding := aes.BlockSize - len(src)%aes.BlockSize
-	padtext := bytes.Repeat([]byte{byte(padding)}, padding)
-	return append(src, padtext...)
+// PKCS7Padding adds padding for any block smaller than AES block size
+func PKCS7Padding(message []byte) []byte {
+	if len(message)%aes.BlockSize == 0 {
+		return message
+	}
+
+	padSize := aes.BlockSize - len(message)%aes.BlockSize
+	addByte := bytes.Repeat([]byte{byte(padSize)}, padSize)
+
+	return append(message, addByte...)
 }
 
-// PKCS7UnPadding unpads as prescribed by the PKCS7 standard
-func PKCS7UnPadding(src []byte) ([]byte, error) {
-	length := len(src)
-	unpadding := int(src[length-1])
+//PKCS7UnPadding removes any padding from plain text
+func PKCS7UnPadding(plainText []byte) ([]byte, error) {
+	plaintextLen := len(plainText)
+	if plaintextLen == 0 {
+		return nil, errors.New("length of plaintext is 0")
+	}
 
-	if unpadding > aes.BlockSize || unpadding == 0 {
-		return nil, fmt.Errorf("invalid padding")
+	removepadSize := int(plainText[plaintextLen-1])
+	if removepadSize == 0 || removepadSize > aes.BlockSize {
```

Comment:
@christo4ferris Are we referring to this panic here? That is to return the plaintext here?